context/src/main/java/io/opentelemetry/context/propagation/ContextPropagatorsBuilder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context.propagation;
+
+/**
+ * A builder of a default {@link ContextPropagators} which can have multiple {@link
+ * TextMapPropagator} added, and extraction and injection will execute every {@link
+ * TextMapPropagator} in order.
+ */
+public interface ContextPropagatorsBuilder {
```

Comment:
Having an interface for a builder, which is just a list of propagators and a constructor call feels like overkill to me. Is the rationale here to make the agent more easily able to bridge the builder? 

Could we ditch the builder entirely and use a pattern like in #2091 where we just have a static method on the ContextPropagators that takes the list of TextMapPropagators?

Reply:
Yup - it came to mind after sending the PR but before also having the idea of killing `ContextPropagators` completely :) Will do so since now I see the binary propagation idea is probably the main reason for this interface.